scan.py

```diff
@@ -197,11 +203,48 @@ def sns_scan_results(
         },
     )
 
+def call_api_with_scan_results(s3_object, result):
+    if AV_STATUS_CALL_API_MAP is None:
+        return
+
+    try:
+        av_status_call_api_map = str(AV_STATUS_CALL_API_MAP)
+        av_status_call_api_map = eval(AV_STATUS_CALL_API_MAP.replace('"', '\"'))
+
+        av_status_call_api = av_status_call_api_map[s3_object.bucket_name]
+
+        av_status_call_api_url = av_status_call_api['url']
+        av_status_call_api_username = av_status_call_api['username']
+        av_status_call_api_password = av_status_call_api['password']
+    except ValueError:
+        print("Skipping API call with scan results, encountered error while parsing AV_STATUS_CALL_API_MAP")
+        return
+
+    message = {
+        "bucket": s3_object.bucket_name,
+        "key": s3_object.key,
+        "version": s3_object.version_id,
+        AV_STATUS_METADATA: result,
+        AV_TIMESTAMP_METADATA: datetime.utcnow().strftime("%Y/%m/%d %H:%M:%S UTC")
+    }
+
+    if av_status_call_api_url is None:
+        print("Skipping API call with scan results, missing url for bucket %s" % s3_object.bucket_name)
+        return
+
+    print("Calling API %s with scan results" % av_status_call_api_url)
+
+    requests.put(
+        av_status_call_api_url,
+        data=message,
+        auth=(av_status_call_api_username, av_status_call_api_password)
+    )
```

Comment:
What if this request fails?

Reply:
We can have the email notification going as of now as suggested, to take care of updating billing with the status with a manual intervention. Since, this is trigger based, there won't be any retrying for sending the request.